plasmapy/atomic/particle_class.py

```diff
@@ -898,6 +862,138 @@ def mass(self) -> u.Quantity:
+        elif self.isotope == 'T':
+            return _special_ion_masses['T 1+']
+        elif self.particle == 'n':
+            return const.m_n
+
+        if not self.isotope:
+            raise InvalidIsotopeError(_category_errmsg(self, 'isotope'))
+
+        base_mass = self._attributes['isotope mass']
+
+        if base_mass is None:  # coveralls: ignore
+            raise MissingAtomicDataError(f"The mass of a {self.isotope} nuclide is not available.")
+
+        _nuclide_mass = self._attributes['isotope mass'] - self.atomic_number * const.m_e
+
+        return _nuclide_mass.to(u.kg)
+
+    @property
+    def mass_energy(self) -> u.Quantity:
+        """
+        Return the mass energy of the particle in joules.
+
+        If the particle is an atom or ion, include the mass of the
+        electrons.
+
+        If the mass of the particle is not known, then raise a
+        `~plasmapy.utils.MissingAtomicDataError`.
+
+        Example
+        -------
+        >>> proton = Particle('p+')
+        >>> proton.mass_energy
+        <Quantity 1.50327759e-10 J>
+        >>> electron = Particle('electron')
+        >>> electron.mass_energy.to('MeV')
+        <Quantity 0.51099895 MeV>
+
+        """
+        try:
+            return (self.mass * const.c ** 2).to(u.J)
+        except MissingAtomicDataError:
+            raise MissingAtomicDataError(
+                f"The mass energy of {self.particle} is not available "
+                f"because the mass is unknown.") from None
+
+    @property
+    def nuclear_mass_energy(self) -> u.Quantity:
+        """
+        Return the mass energy of the nucleus of an isotope.
+
+        If the particle is not an isotope, then raise a
+        `~plasmapy.utils.InvalidIsotopeError`.
+
+        Examples
+        --------
+        >>> protium = Particle('H-1 0+')
+        >>> protium.nuclear_mass_energy
+        <Quantity 1.50327759e-10 J>
+        >>> proton = Particle('p+')
+        >>> proton.mass_energy
+        <Quantity 1.50327759e-10 J>
+
+        """
+        if not self.isotope and not self == 'n':
+            raise InvalidIsotopeError(
+                f"The nuclear mass energy of {self.particle} cannot "
+                f"be found because it is not an isotope. "
+                f"Use mass_energy instead.")
+        return (self.nuclide_mass * const.c ** 2).to(u.J)
+
+    @property
+    def binding_energy(self) -> u.Quantity:
+        """
+        Return the nuclear binding energy in joules.
+
+        This attribute will raise an
+        `~plasmapy.utils.InvalidIsotopeError` if the particle is not a
+        nucleon or isotope.
+
+        Examples
+        --------
+        >>> alpha = Particle('alpha')
+        >>> alpha.binding_energy
+        <Quantity 4.53346938e-12 J>
+        >>> Particle('T').binding_energy.to('MeV')
+        <Quantity 8.48179621 MeV>
+
+        """
+
+        if self._attributes['baryon number'] == 1:
+            return 0 * u.J
```

Comment:
Does this deserve a comment or doctest mention?

Reply:
Added an example in the docstring.